pre_commit/languages/golang.py

```diff
@@ -39,30 +119,36 @@ def install_environment(
         version: str,
         additional_dependencies: Sequence[str],
 ) -> None:
-    helpers.assert_version_default('golang', version)
     env_dir = helpers.environment_dir(prefix, ENVIRONMENT_DIR, version)
 
-    if sys.platform == 'cygwin':  # pragma: no cover
-        gopath = cmd_output('cygpath', '-w', env_dir)[1].strip()
-    else:
-        gopath = env_dir
-    env = dict(os.environ, GOPATH=gopath)
-    env.pop('GOBIN', None)
+    with in_env(prefix, version):
```

Comment:
I had factored this out in your other patch -- this shouldn't be needed here I think

Reply:
Mainly because both hook install and run steps have shared env vars since I kept `go` accessible.

I've refactored it now after fixing #2677.